framework/src/utils/MooseEnumBase.C

```diff
@@ -72,46 +72,24 @@ void
 MooseEnumBase::fillNames(std::string names, std::string option_delim)
 {
   std::vector<std::string> elements;
-  // split on spaces
   MooseUtils::tokenize(names, elements, 1, option_delim);
+  for (const std::string & raw_name : elements)
+    addEnumerationName(raw_name);
+}
 
-  _names.resize(elements.size());
-  int value = 0;
-  for (unsigned int i = 0; i < elements.size(); ++i)
-  {
-    std::vector<std::string> name_value;
-
-    // Make sure the option is not malformed
-    if (elements[i].find_first_of('=') == 0 ||
-        elements[i].find_last_of('=') == elements[i].length() - 1)
-      mooseError("You cannot place whitespace around the '=' character in MooseEnumBase");
-
-    // split on equals sign
-    MooseUtils::tokenize(MooseUtils::trim(elements[i]), name_value, 1, "=");
-
-    if (name_value.size() < 1 || name_value.size() > 2)
-      mooseError("Invalid option supplied in MooseEnumBase: ", elements[i]);
-
-    // See if there is a value supplied for this option
-    // strtol allows for proper conversions of both int and hex strings
-    if (name_value.size() == 2)
-      value = strtol(name_value[1].c_str(), NULL, 0);
-
-    name_value[0] = MooseUtils::trim(name_value[0]);
-
-    // preserve case for raw options, append to list
-    if (i)
-      _raw_names += " ";
-    _raw_names += name_value[0];
-
-    // convert name to uppercase
-    std::string upper(name_value[0]);
-    std::transform(upper.begin(), upper.end(), upper.begin(), ::toupper);
+void
+MooseEnumBase::addEnumerationNames(const std::string & names)
+{
+  fillNames(names);
```

Comment:
I guess I'm trying to understand the reason for the changing API. You added this new method but it just calls an existing method unchanged?